crates/codegen/parser/src/chumsky/character_filter.rs

```diff
@@ -94,13 +92,20 @@ impl<'context> CharacterFilter<'context> {
                 }
             }
 
-            EBNF::DelimitedBy(_)
-            | EBNF::OneOrMore(_)
-            | EBNF::Optional(_)
-            | EBNF::Repeat(_)
-            | EBNF::SeparatedBy(_)
-            | EBNF::Sequence(_)
-            | EBNF::ZeroOrMore(_) => None,
+            // TODO: only do this if the referenced production is marked as inline
+            EBNF::Reference(name) => {
+                if inline_references {
+                    Self::new(
+                        tree,
+                        &tree.context.get_tree_by_name(name).expression(),
+                        inline_references,
+                    )
+                } else {
+                    None
+                }
+            }
+
+            _ => None,
```

Comment:
I suggest keeping the explicit list of "other" variants. I think it helps make sure everytime a variant is updated/added/removed, its effect on this part of the code is updated/reviewed.

Reply:
Sometimes I agree, but this case is specific to selecting case that are applicable to the context i.e. CharacterFilters. The same argument could be made against using the `if let XXX = ...` construct.